igc.h

```diff
@@ -92,6 +133,66 @@ class IgcFormat : public Format
     rec_bad = 1,		// Bad record
   };
 
+  const QHash<QString, IgcFormat::igc_ext_type_t> igc_extension_map{
+    {"ENL", IgcFormat::igc_ext_type_t::ext_rec_enl},
+    {"TAS", IgcFormat::igc_ext_type_t::ext_rec_tas},
+    {"VAT", IgcFormat::igc_ext_type_t::ext_rec_vat},
+    {"OAT", IgcFormat::igc_ext_type_t::ext_rec_oat},
+    {"TRT", IgcFormat::igc_ext_type_t::ext_rec_trt},
+    {"GSP", IgcFormat::igc_ext_type_t::ext_rec_gsp},
+    {"FXA", IgcFormat::igc_ext_type_t::ext_rec_fxa},
+    {"SIU", IgcFormat::igc_ext_type_t::ext_rec_siu},
+    {"ACZ", IgcFormat::igc_ext_type_t::ext_rec_acz},
+    {"GFO", IgcFormat::igc_ext_type_t::ext_rec_gfo},
+  };
+
+  // Will return zero if no match
+  igc_ext_type_t get_ext_type(const QString& type) const {
+    IgcFormat::igc_ext_type_t ret;
+    if (igc_extension_map.contains(type)) {
```

Comment:
This is the bug that lost all the Arrays in the kml output.  You meant to negate contains, or switch the two clauses.  Anyway, with my suggestion https://github.com/GPSBabel/gpsbabel/pull/1054#issuecomment-1496718028 you don't have to check to see if the value is contained at all.